### PR TITLE
Localize dates, counts

### PR DESCRIPTION
**Pull request by estrattonbailey**
Combines #4273 and #4556, cleans up some logic in `useTimeAgo`, and improves translation strings.

### CODE REVIEW

File: src/components/dms/MessageItem.tsx
Comment (estrattonbailey) [inline src/components/dms/MessageItem.tsx:1]: Looks good 👍

File: src/components/forms/DateField/index.shared.tsx [hunk 1]
```diff
@@ -91,7 +92,7 @@ export function DateFieldButton({
             t.atoms.text,
             {lineHeight: a.text_md.fontSize * 1.1875},
           ]}>
-          {localizeDate(value)}
+          {i18n.date(value, {timeZone: 'UTC'})}
```
Comment (estrattonbailey) [inline src/components/forms/DateField/index.shared.tsx:95]: Tested on iOS/Android

File: src/lib/hooks/__tests__/useTimeAgo.test.ts
Comment (estrattonbailey) [inline src/lib/hooks/__tests__/useTimeAgo.test.ts:1]: Split up the diff/format steps of this for easier testing

File: src/lib/hooks/useTimeAgo.ts [hunk 2]
```diff
@@ -1,86 +1,178 @@
...
+    }
+    case 'second': {
+      return long
+        ? i18n._(plural(diff.value, {one: '# second', other: '# seconds'}))
+        : i18n._(
+            defineMessage({
+              message: `${diff.value}s`,
+              comment: `How many seconds have passed, displayed in a narrow form`,
+            }),
+          )
```
Comment (estrattonbailey) [inline src/lib/hooks/useTimeAgo.ts:132]: Great call by Mary in #4273. These show up like this now, which should be nicer for translators.

CleanShot 2024-08-29 at 17 20 31@2x
CleanShot 2024-08-29 at 17 20 25@2x

File: src/lib/strings/time.ts [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
-    day: 'numeric',
-  })} at ${d.toLocaleTimeString(undefined, {
-    hour: 'numeric',
-    minute: '2-digit',
-  })}`
+
+  return i18n.date(d, {
+    dateStyle: 'long',
+    timeStyle: 'short',
+  })
```
Comment (estrattonbailey) [inline src/lib/strings/time.ts:9]: FYI for future use, hardly needs a util 😅

File: src/view/com/util/TimeElapsed.tsx [hunk 4]
```diff
@@ -10,19 +12,21 @@ export function TimeElapsed({
...
-  timeToString?: (timeElapsed: string) => string
+  timeToString?: (i18n: I18n, timeElapsed: string) => string
 }) {
+  const {i18n} = useLingui()
   const ago = useGetTimeAgo()
-  const format = timeToString ?? ago
   const tick = useTickEveryMinute()
   const [timeElapsed, setTimeAgo] = React.useState(() =>
-    format(timestamp, tick),
+    timeToString ? timeToString(i18n, timestamp) : ago(timestamp, tick),
```
Comment (estrattonbailey) [inline src/view/com/util/TimeElapsed.tsx:21]: Kinda wacky to support custom usage in `MessageItem`, but p easy to grok imo

File: src/view/com/util/forms/DateInput.tsx [hunk 5]
```diff
@@ -74,7 +68,9 @@ export function DateInput(props: Props) {
             <Text
               type={props.buttonLabelType}
               style={[pal.text, props.buttonLabelStyle]}>
-              {formatter.format(props.value)}
+              {i18n.date(props.value, {
+                timeZone: props.handleAsUTC ? 'UTC' : undefined,
+              })}
```
Comment (estrattonbailey) [inline src/view/com/util/forms/DateInput.tsx:73]: Tested

File: src/lib/hooks/useTimeAgo.ts [hunk 6]
```diff
@@ -1,86 +1,178 @@
...
   return useCallback(
     (
       earlier: number | string | Date,
       later: number | string | Date,
-      options?: Omit<TimeAgoOptions, 'lingui'>,
+      options?: {format: DateDiffFormat},
     ) => {
-      return dateDiff(earlier, later, {lingui: _, format: options?.format})
+      const diff = dateDiff(earlier, later)
+      return formatDateDiff({diff, i18n, format: options?.format})
```
Comment (estrattonbailey) [inline src/lib/hooks/useTimeAgo.ts:31]: So yeah, split this up for easier testing and readability.

**Review by pfrazee — APPROVED** [review]
LGTM (didnt test)

**Pull request merged**